New resource for Detective Member and Invitation Accepter (#22163)
hashicorp/terraform-provider-aws: website/docs/r/detective_invitation_accepter.html.markdown

---
subcategory: "Detective"
layout: "aws"
page_title: "AWS: aws_detective_invitation_accepter"
description: |-
  Provides a resource to manage an Amazon Detective Invitation Accepter.
---

# Resource: aws_detective_invitation_accepter

> Provides a resource to manage an [Amazon Detective Invitation Accepter](https://docs.aws.amazon.com/detective/latest/APIReference/API_AcceptInvitation.html).

Reviewer: ```suggestion
Provides a resource to manage an [Amazon Detective Invitation Accepter](https://docs.aws.amazon.com/detective/latest/APIReference/API_AcceptInvitation.html). Ensure that the accepter is configured to use the AWS account you wish to _accept_ the invitation from the primary graph owner account.
```
Some clarification around calling accounts.

## Example Usage

```terraform
resource "aws_detective_graph" "primary" {
  provider = "awsalternate"
}

> resource "aws_detective2_member" "primary" {

Reviewer: ```suggestion
resource "aws_detective_member" "primary" {
```

  provider   = "awsalternate"
  account_id = "ACCOUNT ID"
  email      = "EMAIL"

>   graph_arn  = aws_detective_graph.admin.id

Reviewer: ```suggestion
  graph_arn  = aws_detective_graph.primary.id
```

  message    = "Message of the invite"
}

resource "aws_detective_invitation_accepter" "member" {
  graph_arn = aws_detective_graph.primary.id

> }

Reviewer: I especially don't think it makes sense to use the reverse provider setup here in docs.

The example here ought to have only `aws_detective_invitation_accepter` provisioned by `provider = "awsalternate"`.

> resource "aws_detective_invitation_accepter" "member" {
>   graph_arn = aws_detective_graph.primary.id
> }

Reviewer: This needs a `depends_on` `member` statement.
```suggestion
resource "aws_detective_invitation_accepter" "member" {
  graph_arn = aws_detective_graph.primary.id
  depends_on = [aws_detective_member.primary]
}
```

```

## Argument Reference

The following arguments are supported:

* `graph_arn` - (Required) ARN of the behavior graph that the member account is accepting the invitation for.

## Attributes Reference

In addition to all arguments above, the following attributes are exported:

> * `id` - Unique identifier (ID) of the detective invitation accepter.

Reviewer: ```suggestion
* `id` - Unique identifier (ID) of the Detective invitation accepter.
```

## Import

> `aws_detective_invitation_accepter` can be imported using the admin account ID, e.g.

Reviewer: ```suggestion
`aws_detective_invitation_accepter` can be imported using the graph ARN, e.g.
```
From `invitation_accepter.go`: `d.SetId(graphArn)`
